datafusion-cli/src/print_format.rs

```diff
@@ -351,4 +351,82 @@ mod tests {
 
         Ok(())
     }
+
+
+    #[test]
+    fn test_print_batches_empty_batches() -> Result<()> {
+        let batch = RecordBatch::try_from_iter(
+            vec![
+                ("a", Arc::new(Int32Array::from(vec![1, 2, 3])) as ArrayRef),
+            ],
+        )?;
+        let empty_batch = RecordBatch::new_empty(batch.schema());
+
+        PrintBatchesTest::new()
+            .with_format(PrintFormat::Table)
+            .with_batches(vec![
+                empty_batch.clone(),
+                batch,
+                empty_batch,
+            ])
+            .with_expected(
+                &[
+            "+---+",
+            "| a |",
+            "+---+",
+            "| 1 |",
+            "| 2 |",
+            "| 3 |",
+            "+---+\n",
+        ])
+            .run();
+       Ok(())
+    }
+
+    struct PrintBatchesTest {
```

Comment:
I plan to rework the rest of the tests in this module as a follow on PR using this struct

Reply:
https://github.com/apache/arrow-datafusion/pull/8896